backend/notifications/decisionTreeEvaluator.js

```diff
@@ -0,0 +1,51 @@
+const evaluateDecisionTree = (userInterest, notifications) => {
+  const daysSinceLastWorkout = Math.floor(
+    (Date.now() - userInterest.lastWorkoutDate.getTime()) / (1000 * 3600 * 24)
+  );
+
+  const groupedNotifications = {};
+  notifications.forEach((note) => {
+    if (!groupedNotifications[note.type]) {
+      groupedNotifications[note.type] = [];
+    }
+    groupedNotifications[note.type].push(note);
+  });
+
+  if (daysSinceLastWorkout >= 7) {
+    return groupedNotifications["workout_reminder"]?.[0];
+  }
+
+  if (userInterest.activityScore < 30) {
+    return groupedNotifications["motivation"]?.[0];
+  }
+
+  if (groupedNotifications["achievement"]) {
+    return groupedNotifications["achievement"][0];
+  }
+
+  if (groupedNotifications["social_interaction"]) {
+    return groupedNotifications["social_interaction"][0];
+  }
+
+  let topInterest = { score: 0 };
```

Comment:
would be good to add a comment summarizing this logic. so we'll only send them a template recommendation if they are extremely interested in a particular category?